src/map/utils/petutils.cpp

```diff
@@ -817,6 +820,74 @@ namespace petutils
         }
     }
 
+    void SpawnTrust(CCharEntity* PMaster, uint32 TrustID)
+    {
+        // TODO: You can only spawn trusts in battle areas, similar to pets. See MSGBASIC_TRUST_NOT_HERE
+
+        // TODO: There is an expandable limit of trusts you can summon, based on key items.
+        size_t maxTrusts = 5;
+
+        // TODO: These checks should be done at before spellcast time!!
+        // If you're in a party, you can only spawn trusts if:
+        //  * You're the party leader
+        //  * The party isn't full
+        //  * The party isn't part of an alliance
+        if (PMaster->PParty != nullptr)
+        {
+            CBattleEntity* leader = PMaster->PParty->GetLeader();
+            if (leader == nullptr || leader->id != PMaster->id)
+            {
+                PMaster->pushPacket(new CMessageStandardPacket(PMaster, 0, MSGSTD_TRUST_SOLO_OR_LEADER));
+                return;
+            }
+            if (PMaster->PParty->members.size() >= 6)
+            {
+                PMaster->pushPacket(new CMessageStandardPacket(PMaster, 0, MSGSTD_TRUST_LIMIT));
+                return;
+            }
+            if (PMaster->PParty->m_PAlliance != nullptr)
+            {
+                PMaster->pushPacket(new CMessageStandardPacket(PMaster, 0, MSGSTD_TRUST_SOLO_OR_LEADER));
+                return;
+            }
+
+            // Reduce the max number of summonable trusts
+            maxTrusts = 6 - PMaster->PParty->members.size();
+        }
+
+        if (PMaster->PTrusts.size() >= maxTrusts)
+        {
+            PMaster->pushPacket(new CMessageStandardPacket(PMaster, 0, MSGSTD_TRUST_LIMIT));
+            return;
+        }
+
+        // You can't spawn the same trust twice
+        // TODO: This includes otherwise distinct trusts, e.g. Shantotto and Shantotto II, only 1 can be called.
+        //       It'd probably be "good enough" to use the name as a heuristic, looking for "II" (this catches 99% of them).
+        for (auto trust : PMaster->PTrusts) {
+            if (trust->m_PetID == TrustID) {
+                PMaster->pushPacket(new CMessageStandardPacket(PMaster, 0, MSGSTD_TRUST_SAME));
+                return;
+            }
+        }
+
+        // Make a new party if we weren't in one.
+        // TODO: It's actually not a real party: /sea shows your name as grey not yellow, but it shows as a party on the GUI.
+        if (PMaster->PParty == nullptr)
+        {
+            PMaster->PParty = new CParty(PMaster);
+        }
+
+        CTrustEntity* PTrust = LoadTrust(PMaster, TrustID);
+        PMaster->PTrusts.insert(PMaster->PTrusts.begin(), PTrust);
+        PMaster->StatusEffectContainer->CopyConfrontationEffect(PTrust);
+        PMaster->loc.zone->InsertPET(PTrust);
+        PMaster->PParty->ReloadParty();
+
+        // Tell the client this is a trust (shows the 'Release' option on the menu).
+        PMaster->pushPacket(new CTrustSyncPacket(PMaster, PTrust));
```

Comment:
do we need to push this packet here? can it work just by pushing the trust sync packet in the trusts AI tick?

Reply:
The AI ticks don't currently work properly, so yes this is intended. This also matches `SpawnPet` which does something similar.